app/controllers/api/v1/accounts/featured_tags_controller.rb

```diff
@@ -3,10 +3,14 @@
 class Api::V1::Accounts::FeaturedTagsController < Api::BaseController
   before_action :set_account
   before_action :set_featured_tags
+  skip_before_action :require_authenticated_user!, only: [:index]
 
   respond_to :json
 
   def index
+    if user_would_block_unauthenticated_api_access?(@account)
+      user_blocks_unauthenticated_api_access and return
+    end
```

Comment:
```suggestion
    if user_would_block_unauthenticated_api_access?(@account)
      user_blocks_unauthenticated_api_access and return
    end
```

Since you end up having this statement in a bunch of your controllers, you could move it up to the `Api::BaseController` and implement it once.

Reply:
My very basic Ruby grasp is what led me to doing this this way. I needed `user_blocks_unauthenticated_api_access` to return the JSON object and 401 status, and I wasn't sure how to invoke the render behavior and return from the function. Earlier iterations of this change set had a method that called this, but the calling code would continue and would try to render another result, which resulted in errors in the logs as a response was already being sent. The `... and return` pattern was the only way I could think of to solve this, but I'm very open to alternatives (and to learning!)

Reply:
I think the `and return` approach would work still work if you moved this up to a method in the base controller. But maybe not. Controller behavior in Rails is frequently surprising.